internal/users/manager.go

```diff
@@ -257,14 +257,8 @@ func (m *Manager) updateUser(u types.UserInfo) (err error) {
 		return err
 	}
 
-	lockedGroups, cleanup, err := localentries.GetGroupsWithLock()
-	if err != nil {
-		return err
-	}
-	defer func() { err = errors.Join(err, cleanup()) }()
-
```

Comment:
No, sorry it's better if we don't do this... As that's the whole reason for having a ref-counted locking is that we don't want to do a lock/unlock dance to prevent races.

So, while in this specific case it may not happen because another lock may be set already by the fact we've new groups or new users, it's not a thing it may be true... So I'd prefer to keep the lock around until `UpdateUser` is completed, if we ever acquire it

Reply:
I think you misunderstand. The call to `GetGroupsWithLock` was just moved to the new `localentries.UpdateGroups` function.

Reply:
why? it's not doing anything that requires a lock on the userdb after calling `localentries.UpdateGroups`

Reply:
and if we change the code and do add more operations which require a lock we can call `localentries.GetGroupsWithLock` instead and handle the unlocking in `users.updateUser`

Reply:
Not particularly (a part the fact that we act on the user folder, and we don't really want someone else to modify the user dir, until we're not done with it), but we could do even more in future and I want to be protected from that. That was why I preferred to rely on defer, they only run on return...

Getting and releasing the lock is bad, and risky since things may change meanwhile, while I'd prefer to ensure that the state sticks to the one we had when starting the transaction.